src/lib.rs

```diff
@@ -716,15 +716,52 @@ pub mod unsync {
     /// //   92
     /// ```
     pub struct Lazy<T, F = fn() -> T> {
-        cell: OnceCell<T>,
-        init: Cell<Option<F>>,
+        state: Cell<State>,
+        data: Data<T, F>,
```

Comment:
Note that we could make `Lazy` even more compact by moving `state` at the beginning of each union variant (and making the `union` be `#[repr(C)]`. If `T = [u8; 3]`, and `F = u16`, this would make `Lazy<T, F>` be `4 bytes` rather than `6`, for instance.